### ISSUE TITLE
Spinlock staring because fail to update requirements

### ISSUE THREAD

**Issue by Samonitari** [esoadamo/simple-guardian#2]
Hi!

After installation, simple giardian produces this constanlty.
I presume the `Restart=on-failure` makes this a repeating trial-and-failure

BTW, why add user to the adm group (which is non-uniformly-existing across distros), I had to comment it out from install.py.

```
journalctl -u simple-guardian
...
Mar 22 08:31:41 xyx python[15299]: new requirements are present, updating...
Mar 22 08:31:41 xyz python[15299]: !!!!! updating requirements failed!!!
Mar 22 08:31:41 xyz python[15299]: ERROR: Can not perform a '--user' install. User site-packages are not visible in this virtualenv.
Mar 22 08:31:42 xyz python[15299]: WARNING: You are using pip version asdf; however, version asdf is available.
Mar 22 08:31:42 xyz python[15299]: You should consider upgrading via the '/usr/share/simple-guardian/venv/bin/python -m pip install --upgrade pip' command.
...
```

**Comment by esoadamo** [comment]
Hi,
thank's a lot for reporting this! It is a great feeling to see that someone uses something that you have created!
Unfortunately, as much as I love simple guardian, it was a project that I created when I still was in high school and now I see how much flawed this applications is (not very flexible code base, some bad practices).
With that in mind and that I am currently very busy, I am unable to fix this issue.

But if you think that you can provide a PR with a fix, I 100% encourage you to do so! I will gladly merge it and add you to a list of contributors!

Second option is to wait until I release simple-guardian 3.0 (internally called by me `sg ad-hoc`), this version will follow better practices and philosophy and will be heavily focused on performance and easy integration. (You can watch my progress in the [public repo](https://github.com/esoadamo/simple-guardian-ad-hoc))

tl;dr
Thanks a lot, it means so much to me to see that all that work is not only for myself, but I don't have time to fix it now, but still thanks a lot lot lot!

**Comment by Samonitari** [comment]
Hi!

> Unfortunately, as much as I love simple guardian, it was a project that I created when I still was in high school and now I see how much flawed this applications is (not very flexible code base, some bad practices).

Wow. Keep it going, and don't burn out! Although not unreasonably complicated is the problem, but knowing enough about python programming, linux, systemd, iptables, ssh etc. in high school is very formidable!

The reason I tried your project is I won't run anything dependent on python2 on my server, as any program worth its salt and actively maintained should be alread ported (and I like a lean system so I won't install any python2 package for any exception). Fail2ban just fails here.

Not sure if I'll create a pull request.
But I'll watch your successor project closely, and test it when it's ready.
And most probably I'll be a bit nitpicky!

Best wishes!
